packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -37,10 +37,28 @@ import {
 } from './gridVirtualizationSelectors';
 import { EMPTY_RENDER_CONTEXT } from './useGridVirtualization';
 
+const EMPTY_SCROLL_POSITION = { top: 0, left: 0 };
+
 export const EMPTY_DETAIL_PANELS = Object.freeze(new Map<GridRowId, React.ReactNode>());
 
 export type VirtualScroller = ReturnType<typeof useGridVirtualScroller>;
 
+enum ScrollDirection {
+  NONE,
+  UP,
+  DOWN,
+  LEFT,
+  RIGHT,
+}
+
+const EMPTY_SCROLL_CACHE = {
+  direction: ScrollDirection.NONE,
+  lastTimestamp: -1,
+  lastPosition: { top: 0, left: 0 },
+  buffer: { top: 0, left: 0 },
+};
+type ScrollCache = typeof EMPTY_SCROLL_CACHE;
```

Comment:
The general idea for the logic is that scroll events trigger right after a new frame has been rendered, so too late for us to prevent white areas that way. If we want to prevent white areas, we need to diplay enough rows/columns so that there is enough of them to cover the area from the current viewport to where the viewport will be at the next scroll event.

As mentioned above predicting by how much the user will have scrolled at the next scroll event is a problem is itself, but by having a substantially large buffer in one direction we try to ensure users see as few white areas as possible.